### PR TITLE
feat(v2 backend): Reduce Tekton loop graph complexity with new design

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Part of #1419 

**Description of your changes:**

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/src/v2/compiler/tektoncompiler/tekton.go [hunk 1]
```diff
@@ -481,6 +482,24 @@ func (c *pipelinerunCompiler) PushLoop(loop *pipelineloopapi.PipelineLoop) {
...
+	if c.loopNames == nil {
+		c.loopNames = make(map[string]string)
+	}
+	if name == "" {
+		return
+	}
+	c.loopNames[name] = "true"
+}
+
+func (c *pipelinerunCompiler) GetLoopName(name string) bool {
```
Comment (yhwang) [inline backend/src/v2/compiler/tektoncompiler/tekton.go:495]: `HasLoopName` seems better

Comment (Tomcli) [reply backend/src/v2/compiler/tektoncompiler/tekton.go]: Done

**Review by yhwang — APPROVED** [review]
/lgtm

**Pull request merged**